src/link/Coff.zig

```diff
@@ -2460,18 +2437,23 @@ pub fn getGlobalSymbol(coff: *Coff, name: []const u8, lib_name_name: ?[]const u8
     gop.value_ptr.* = sym_loc;
 
     const gpa = coff.base.comp.gpa;
-    const sym = coff.getSymbolPtr(sym_loc);
-    try coff.setSymbolName(sym, name);
-    sym.storage_class = .EXTERNAL;
-
-    if (lib_name_name) |lib_name| {
-        // We repurpose the 'value' of the Symbol struct to store an offset into
-        // temporary string table where we will store the library name hint.
-        sym.value = try coff.temp_strtab.insert(gpa, lib_name);
+    const symbol = coff.getSymbolPtr(sym_loc);
+    try coff.setSymbolName(symbol, symbol_name);
+    symbol.storage_class = .EXTERNAL;
+
+    if (lib_name) |name| {
+        // We don't yet know if the symbol will be resolved in an object or a DLL,
+        // so pre-emptively create an Import for this symbol.
+        // TODO the question here is though, should we return an error if the symbol is
+        // resolved not in a library?
```

Comment:
Let's solve this problem rather than writing a TODO comment. 

This function is called during `updateNav`. That means there are two possible solutions:

1. Organize the pipeline to finish the "pre-link" phase, before processing any ZigObject updates, in which case this information will always be known.
2. Support handling `updateNav` interleaved with all the other linker inputs being processed, storing whatever information is needed to finish the link later. In this case, there should be no TODO question here. Store whatever information is necessary to make linking equivalent to (1).

Personally I like (1) the best since it optimizes for doing the least amount of work during incremental updates in which only the zig source code is changed, as well as for simplicity, particularly with what data is stored in the linker.